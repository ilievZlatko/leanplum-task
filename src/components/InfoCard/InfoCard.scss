@import '../../styles/colors';
@import '../../styles/spacings';
@import '../../styles/fontSizes';
@import '../../styles/mixins';

.info-card {
	background-color: $color-title-bg;
	@include border-radius;
	padding: $space-xl;

	&__title {
		color: $color-subtitle;
		margin-bottom: $space-xl;
	}

	&__content {
		color: $color-dark-text;
		font-size: 28px;
		font-weight: bold;
	}
}
