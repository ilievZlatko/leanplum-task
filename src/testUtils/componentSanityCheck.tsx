import React from 'react'
import { shallow } from 'enzyme'

export const componentSanityCheck = (
	label: string,
	Element: React.ReactElement,
) =>
	describe(`Sanity checks for ${label}`, () => {
		it('is defined', () => {
			expect(Element).toBeDefined()
		})

		it('renders the element', () => {
			const wrapper = shallow(Element)
			const expected = false
			const actual = wrapper.isEmptyRender()
			expect(expected).toEqual(actual)
		})

		it('matches the snapshot', () => {
			const wrapper = shallow(Element)
			const expected = wrapper.html()
			expect(expected).toMatchSnapshot()
		})
	})
