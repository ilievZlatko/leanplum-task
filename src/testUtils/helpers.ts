import { ShallowWrapper, ReactWrapper } from 'enzyme'

export const findByTestAttr = (
	component: ShallowWrapper | ReactWrapper | any,
	attr: string,
): ShallowWrapper | ReactWrapper => {
	return component.find(`[data-test='${attr}']`)
}
