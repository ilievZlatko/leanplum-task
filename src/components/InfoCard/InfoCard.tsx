import React from 'react'

import './InfoCard.scss'

interface Props {
	title: string
	content: string | React.ReactNode
}

const InfoCard: React.FC<Props> = ({ title, content, ...otherProps }) => {
	return (
		<div className='info-card' {...otherProps}>
			<h2 data-test='info-card-title' className='info-card__title'>
				{title}
			</h2>
			{typeof content === 'string' ? (
				<div data-test='info-card-content' className='info-card__content'>
					{content}
				</div>
			) : (
				<div data-test='info-card-content'>{content}</div>
			)}
		</div>
	)
}

export default InfoCard
