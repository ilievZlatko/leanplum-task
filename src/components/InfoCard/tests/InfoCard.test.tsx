import React from 'react'
import { shallow } from 'enzyme'

import InfoCard from '../InfoCard'
import { componentSanityCheck } from '../../../testUtils/componentSanityCheck'
import { findByTestAttr } from '../../../testUtils/helpers'

describe('InfoCard', () => {
	componentSanityCheck(
		'InfoCard',
		<InfoCard title='test title' content='test content' />,
	)

	it('Should accept title and content of type ReactNode', (done: jest.DoneCallback) => {
		const wrapper = shallow(
			<InfoCard
				title='Info card title'
				content={
					<div>
						<span>line one</span>
						<span>line two</span>
					</div>
				}
			/>,
		)

		expect(wrapper.html()).toContain(
			'<div><span>line one</span><span>line two</span></div>',
		)
		done()
	})

	it('Should accept title and content of type string', (done: jest.DoneCallback) => {
		const wrapper = shallow(
			<InfoCard title='Info card title' content='Info card content' />,
		)
		const content = findByTestAttr(wrapper, 'info-card-content')

		expect(content.text()).toEqual('Info card content')
		done()
	})
})
