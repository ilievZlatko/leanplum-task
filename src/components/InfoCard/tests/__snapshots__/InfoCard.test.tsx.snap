// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Sanity checks for InfoCard matches the snapshot 1`] = `"<div class=\\"info-card\\"><h2 data-test=\\"info-card-title\\" class=\\"info-card__title\\">test title</h2><div data-test=\\"info-card-content\\" class=\\"info-card__content\\">test content</div></div>"`;
